src/aggregator/client/conn.go

```diff
@@ -152,6 +158,34 @@ func (c *connection) Close() {
 	c.mtx.Unlock()
 }
 
+func (c *connection) upgradeToTLS(conn net.Conn) (net.Conn, error) {
+	certPool := x509.NewCertPool()
+	if c.tls.CAFile() != "" {
+		certs, err := os.ReadFile(c.tls.CAFile())
```

Comment:
Should we be caching the cert file? We recreate connections with some amount of frequency.

Reply:
Or maybe more appropriately, it just goes in the pool.

Reply:
I have added a TTL config cache on the connection level. It will be helpful for reconnecting but not for establishing new connections. The client has fewer connections than the server, so I believe it will be a neglectable overhead.

Reply:
Sure, fair enough. Just for my understanding though--why not use the same cache for connecting as well?

Reply:
That's because the write manager is responsible for creating connections(it can be considered as a connection pool). Theoretically, it can contain TLS options and use its TLS config cache to establish a connection. However, it knows nothing about connection parameters, and I don't want it to have that information because it doesn't seem to be something that should be responsible for secure connections. Also, even if it would manage the TLS configuration, a connection should do certificate rotation on its own in case of reconnection.